src/wallet/wallet.cpp

```diff
@@ -4560,31 +3942,141 @@ std::vector<OutputGroup> CWallet::GroupOutputs(const std::vector<COutput>& outpu
     return groups;
 }
 
-bool CWallet::GetKeyOrigin(const CKeyID& keyID, KeyOriginInfo& info) const
+bool CWallet::SetCrypted()
 {
-    CKeyMetadata meta;
+    LOCK(cs_wallet);
+    if (fUseCrypto)
+        return true;
+    fUseCrypto = true;
+    return true;
```

Comment:
Is this done as an optimization? I'd prefer:
```
void CWallet::SetCrypted()
{ 
    LOCK(cs_wallet);
    fUseCrypto = true;
}
```

Reply:
This is moved code and I will not be changing it in order to avoid overburdening reviewers with ensuring that behavior hasn't changed.